src/main/java/net/unaussprechlich/warlordsplus/module/modules/FlagTakenDetector.kt

```diff
@@ -0,0 +1,43 @@
+package net.unaussprechlich.warlordsplus.module.modules
+
+import net.minecraft.client.Minecraft
+import net.minecraftforge.client.event.ClientChatReceivedEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import net.unaussprechlich.eventbus.EventBus
+import net.unaussprechlich.eventbus.IEvent
+import net.unaussprechlich.warlordsplus.module.IModule
+import net.unaussprechlich.warlordsplus.util.contain
+import net.unaussprechlich.warlordsplus.util.removeFormatting
+
+object FlagTakenDetector : IModule {
+
+    @SubscribeEvent
+    fun onChatMessage(e: ClientChatReceivedEvent) {
```

Comment:
Detecting it by chat messages  renders the flag detection unreliable. Instead you should just loop over the already existing tab list and see who has the flag.

Use `Minecraft.getMinecraft().thePlayer!!.sendQueue.playerInfoMap` instead and loop over it.  The flag symbol should then be in `it.playerTeam.colorSuffix`.

Reply:
whats the symbol?

Reply:
Idk, you probably have to use the Debugger to find out